lib/src/tree_cursor.c

```diff
@@ -172,6 +172,7 @@ void ts_tree_cursor_init(TreeCursor *self, TSNode node) {
     .child_index = 0,
     .structural_child_index = 0,
     .descendant_index = 0,
+    .root_alias = node.context[3],
```

Comment:
I don't understand why we need this `root_alias` field. Nodes can only be aliased by their parent node having an `alias_sequence` (which indicate the alias of every child), and we already account for that. Can you explain why the `root_alias` is there?

Reply:
Actually, re-reading your comment about calling `ts_tree_cursor_init` about a non-root, aliased node, I get the intent now. I don't think `root_alias` should be on *every* node of the stack though. It's just the alias for the node at the bottom of the stack right?

Reply:
yep. it's *only* used when the stack size is just 1, sorry if the else if bit wasn't clear about that